metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/OrchestratorService.java

```diff
@@ -726,6 +743,35 @@ public DatasetExecutionInformation getDatasetExecutionInformation(MetisUser meti
     return datasetExecutionInformation;
   }
 
+  private void validateAndTrimHarvestParameters(Workflow workflow)
+      throws MalformedURLException, URISyntaxException {
+    OaipmhHarvestPluginMetadata oaipmhPluginMetadata = (OaipmhHarvestPluginMetadata) workflow
+        .getPluginMetadata(PluginType.OAIPMH_HARVEST);
+    if (oaipmhPluginMetadata != null) {
+      URL u = new URL(oaipmhPluginMetadata.getUrl().trim()); // this would check for the protocol
+      u.toURI(); // does the extra checking required for validation of URI
+      //Remove all the query parameters
+      int queryIndexStart = oaipmhPluginMetadata.getUrl().indexOf('?');
+      String urlWithoutQueryParameters = queryIndexStart >= 0 ? oaipmhPluginMetadata.getUrl()
+          .substring(0, oaipmhPluginMetadata.getUrl().indexOf('?'))
+          : oaipmhPluginMetadata.getUrl();
+      oaipmhPluginMetadata.setUrl(urlWithoutQueryParameters.trim());
```

Comment:
I feel it's typically better to use libraries to mess with URIs. In this case, the Java API itself can remove the query (and supposedly the fragment too):
`String strippedUri = new URI(uri.getScheme(), uri.getAuthority(), uri.getPath(), null, null).toString();`

Reply:
okay that looks better indeed.